lib/Geography/MapAlgebra.hs

```diff
@@ -241,27 +239,13 @@ instance Projection WebMercator where
 -- @
 newtype Raster u p (r :: Nat) (c :: Nat) a = Raster { _array :: Array u Ix2 a }
 
-{-
-instance Show a => Show (Raster p r c a) where
-  show (Raster a) = ('\n' :) . unlines . map unwords $ groupsOf cols padded
-    where (Z :. r :. c) = R.extent a
+-- | Warning: This will evaluate (at most) the 10x10 top-left corner of your
+-- `Raster` for display. This should only be used for debugging.
+instance (Show a, Load (EltRepr u Ix2) Ix2 a, Size u Ix2 a) => Show (Raster u p r c a) where
+  show (Raster a) = show . computeAs B $ extract' (0 :. 0) (rows :. cols) a
+    where (r :. c) = size a
```

Comment:
By the way, you can do all sort of cool things with index too, eg:
```
(rows :. cols) = A.liftIndex (P.min 10) $ size a
```
It isn't advertised too much in the package doc, but there are some index goodies here: https://www.stackage.org/haddock/lts-11.3/massiv-0.1.4.0/Data-Massiv-Core-Index.html#t:Index

Reply:
Oh hot damn, I didn't realize there were so many cool toys in there. `zeroIndex`!

Reply:
Yeah, `Ix n` is also a `Num`, so you can do things like `(3 :. 4) + 1 == (4 :. 5)`. So instead of `zeroIndex`, you could also do `fromIntegral 0` :)

Reply:
Shenanigans!